ci-operator/config/openshift/cluster-api-provider-agent/openshift-cluster-api-provider-agent-master.yaml

```diff
@@ -61,6 +95,36 @@ tests:
   container:
     from: src
   skip_if_only_changed: (^(docs|config)/)|(\.md$)|((^|/)OWNERS$)
+- as: e2e-capi-provider-hypershift
+  optional: true
+  steps:
+    cluster_profile: packet-assisted
+    env:
+      ASSISTED_CONFIG: |
+        KUBE_API=yes
+        ENABLE_KUBE_API=true
+        MINIKUBE_DISK_SIZE=70g
+        MINIKUBE_RAM_MB=10240
+        TEST=./src/tests/test_kube_api.py
+        TEST_FUNC=test_capi_provider
+        MAKEFILE_TARGET="create_full_environment run deploy_capi_env test"
+      TEST_SUITE: minimal
+    workflow: baremetalds-assisted-install
+- as: e2e-capi-provider-hypershift-periodic
+  cron: '@daily'
+  steps:
+    cluster_profile: packet-assisted
+    env:
+      ASSISTED_CONFIG: |
+        KUBE_API=yes
+        ENABLE_KUBE_API=true
+        MINIKUBE_DISK_SIZE=70g
+        MINIKUBE_RAM_MB=10240
+        TEST=./src/tests/test_kube_api.py
+        TEST_FUNC=test_capi_provider
+        MAKEFILE_TARGET="create_full_environment run deploy_capi_env test"
+      TEST_SUITE: minimal
```

Comment:
nit, usually we do:
```suggestion
      TEST_SUITE: full
```

Reply:
the test function installs a hypershift cluster, it can pass the minimal suite.

Reply:
So it probably can pass the full suite, no?

Reply:
The reasoning is to have more conformance tests on periodic jobs

Reply:
It can't, when I tried to run it with minimal (different test suite than the test suite defined in )
`./openshift-tests run openshift/conformance/parallel --dry-run | grep minimal | ./openshift-tests run -f - --junit-dir tests `
it failed in all these:
```
[Conformance][sig-api-machinery][Feature:APIServer] local kubeconfig "lb-ext.kubeconfig" should be present on all masters and work [Suite:openshift/conformance/parallel/minimal]
[Conformance][sig-api-machinery][Feature:APIServer] local kubeconfig "lb-int.kubeconfig" should be present on all masters and work [Suite:openshift/conformance/parallel/minimal]
[Conformance][sig-api-machinery][Feature:APIServer] local kubeconfig "localhost-recovery.kubeconfig" should be present on all masters and work [Suite:openshift/conformance/parallel/minimal]
[Conformance][sig-api-machinery][Feature:APIServer] local kubeconfig "localhost.kubeconfig" should be present on all masters and work [Suite:openshift/conformance/parallel/minimal]
[sig-arch] [Conformance] FIPS TestFIPS [Suite:openshift/conformance/parallel/minimal]
[sig-cli] Kubectl client Kubectl cluster-info should check if Kubernetes control plane services is included in cluster-info  [Conformance] [Suite:openshift/conformance/parallel/minimal] [Suite:k8s]
[sig-network-edge][Conformance][Area:Networking][Feature:Router] The HAProxy router should be able to connect to a service that is idled because a GET on the route will unidle it [Skipped:Disconnected] [Suite:openshift/conformance/parallel/minimal]
[sig-network-edge][Conformance][Area:Networking][Feature:Router] The HAProxy router should pass the gRPC interoperability tests [Suite:openshift/conformance/parallel/minimal]
[sig-network-edge][Conformance][Area:Networking][Feature:Router] The HAProxy router should pass the h2spec conformance tests [Suite:openshift/conformance/parallel/minimal]
[sig-network-edge][Conformance][Area:Networking][Feature:Router] The HAProxy router should pass the http2 tests [Suite:openshift/conformance/parallel/minimal]
```
HyperShift is a hosted control plane with no master nodes, and some tests seem to care about it.
For more details, you can look at the comments on this [Jira](https://issues.redhat.com/browse/MGMT-8602).
In any case, the provider repo doesn't really care about HyperShift conformance, it cares about the setup phase that add nodes to the Hypershift cluster.